pkg/ui/workspaces/cluster-ui/src/sqlActivity/util.tsx

```diff
@@ -151,14 +154,24 @@ export const convertRawStmtsToAggregateStatistics = (
       database: stmt.database,
       applicationName: stmt.app,
       stats: stmt.stats,
+      diagnosticsReports: (statementDiagnostics || []).filter(
+        d => d.statement_fingerprint === stmt.statement,
+      ),
     };
   });
 };
 
 // We separate the memoized version for testing.
 export const convertRawStmtsToAggregateStatisticsMemoized = createSelector(
-  (stmts: CollectedStatementStatistics[]) => stmts,
-  (stmts): AggregateStatistics[] => convertRawStmtsToAggregateStatistics(stmts),
+  (
+    stmts: CollectedStatementStatistics[],
+    statementDiagnostics: StatementDiagnosticsResponse,
```

Comment:
How often will we be polling for stmt diags? If it's pretty often, can we move the correlation logic of the diag report with the stmt fingerprint to outside of this memoized function (and the above convertRaw function). Otherwise we'll lose the benefit of this caching if we're getting a new stmt diag response every 10s or so.

Reply:
Right now it requests fingerprints every `5m`, but will probably be more frequent in scope of https://github.com/cockroachdb/cockroach/issues/106154 cahnge